File: consultorio-frontend/src/app/models/horario-atencion.model.ts
export interface HorarioAtencion {
  horarioAtencionId: number;
  inicioAtencion: string;
  finAtencion: string;
  inicioFeriado?: string | null;
  finFeriado?: string | null;
}

export interface HorarioAtencionPayload {
  inicioAtencion: string;
  finAtencion: string;
  inicioFeriado?: string | null;
  finFeriado?: string | null;
}

export interface ActualizarHorarioRequest {
  horario: HorarioAtencionPayload;
}

File: consultorio-backend/Models/HorarioAtencionModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace ConsultorioMedico.API.Models
{
    public class HorarioAtencionModel
    {
        [Key]
        public int HorarioAtencionId { get; set; }

        [Required]
        public DateTime InicioAtencion { get; set; }

        [Required]
        public DateTime FinAtencion { get; set; }

        public DateTime? InicioFeriado { get; set; }

        public DateTime? FinFeriado { get; set; }
    }
}
